### DISCUSSION TITLE
Record actions on custom page

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#2873] [Help] [answered]
I`ve got categories resource and because they can be few levels deep and I want to have clear view of ancestors, I resigned from table view to recursively show categories tree:

![categories](https://user-images.githubusercontent.com/3818874/174842524-b148efc6-f948-4fa8-9307-d2f89a2bd90f.jpg)

To create that I use simple (modal) resource with my own view:

```blade
<x-filament::page class="filament-resources-list-records-page">

  @include('filament.resources.categories.partial.subcategory-list', ['categories' => $categories])

</x-filament::page>
```

Now, what I`m trying to do is to make actions which works for given record in ```<ul>``` list: 
1. **Add subcategory** - The same form as create action, but with prepopulated parent field. Ideally if it can reuse create form. Maybe replicate action can come in handy (?)
2. **Delete category**

The problem is that if I register this actions via method ```getActions()``` they will appear next to "Create" button and doesnt work in context of given category. Maybe I could hide them from there, but I don`t know if there is any possibility to call it with passing record.

I could also register them with ```getTableActions()```, but then I should have table on page, so modals will be inserted on page and accessible. I could insert table and hide it, but this seems very inelegant. Using this approach the best solution will be to insert in view only component which mounts action modals, but I couldn`t figure this out.

Is something I`m missing or any another approach?

**Comment by danharrin** [comment, marked as answer]
Actions only work in designated places, it's best if you just use a custom `<x-filament::modal>` with the form builder and Livewire.

**Reply by studiowizjo** [reply]
Yes, I could do that, but finally I figured out how to do this exactly as I want, benefiting from using table actions. Here is the solution:

1. Add to custom component page desired actions using `getTableActions()`
2. Create custom view and to it some code responsible for mounting actions, stealed from **filament\tables\resources\views\index.blade.php**, with 2 small changes:

```blade
{{-- this comes from: vendor\filament\tables\resources\views\index.blade.php --}}
<form wire:submit.prevent="callMountedTableAction">
@php
    $action = $this->table->getMountedAction(); //1st change
@endphp

<x-tables::modal :id="\Illuminate\Support\Str::of(static::class)->replace('\\', '\\\\') . '-table-action'" :visible="filled($action)" :width="$action?->getModalWidth()" display-classes="block">
    @if ($action)
        //...

        @if ($action->hasFormSchema())
            {{-- 2nd change --}}
            {{ $this->table->getMountedActionForm() }}
        @endif

        //...
    @endif
</x-tables::modal>
</form>
```
